### ISSUE TITLE
Bot Randomly Stops Playing Music

### ISSUE THREAD

**Issue by Vanituhs** [Just-Some-Bots/MusicBot#263]
Hey there,

I don't know if anyone else is experiencing this bug, but occasionally the bug will randomly stop playing music, and it produces **zero errors**, and **zero console errors**. I think it may have something to do with the connection, but I'm not sure honestly because my voice chat (normally) works perfectly fine, and my internet connection is generally strong to the servers (less than 50ms ping).

Does anyone know what could be the cause for this? Typing (!restart) in the chat seems to work, but we lose the entire queue of songs and everything else, so that doesn't help us very much truthfully, if anything, we'd just like the problem to resolve. It doesn't have a time limit, sometimes the bot will last all night (over 15 hours), and sometimes it'll disconnect in literally 30 minutes. It's all random.

Any help on this at all? What can I change, add or do?

**Comment by imayhaveborkedit** [comment]
Its because the websocket got closed, most likely by cloudflare.  Known issue, you should just be able to `!disconnect` `!summon` it back.  I'm trying to figure out how to fix this properly, its not a simple fix.

**Comment by narrundo** [comment]
I'm having same issue with no error in my logs. And sometimes musicbot didn't make any sound although its already downloading and queue the song

**Comment by imayhaveborkedit** [comment]
There are a couple ways to deal with this right now.  Restarting the bot, obviously, but you lose the your queue.  Same with the `!restart` command.  You should be able to `!disconnect` and `!summon` it back.  Changing the server region should also make it reconnect.

**Comment by Vanituhs** [comment]
Hey @imayhaveborkedit, I was just wondering if any progress has been made on a solution for this issue? I've found that using both "!restart" (as a stand-alone command), or using "!disconnect" and waiting till the bot leaves the channel, then clicking "!summon" works as well, but that does get annoying when you're in the middle of a long game of League and you have to pause it and close down your client just for something like that.

I know it's not a simple problem to fix, but was just wondering if you'd made any progress at all on something like this.

**Comment by imayhaveborkedit** [comment]
I added 9109dcd535a82a86743813071cc8be0f0ee1d620 a little while ago.  It seems to work when I used my bot earlier.

**Comment by Vanituhs** [comment]
Where do I add that? @imayhaveborkedit

**Comment by imayhaveborkedit** [comment]
You use the latest version of the bot.

**Comment by adamja** [comment]
Did this problem definitely get fixed? I've got 2 bots running in my discord server and they both appear to disconnect randomly, basically exactly like the initial post of this thread described. Everything else appears to work perfectly. 
I only set them up a couple of days ago, so they should be at the latest version (following [this](https://github.com/Just-Some-Bots/MusicBot/wiki/Guide-for-Raspbian) guide).
